src/NuGet.Clients/NuGet.SolutionRestoreManager/ProjectRestoreInfo3Adapter.cs

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using NuGet.ProjectManagement;
+
+namespace NuGet.SolutionRestoreManager
+{
+    internal record ProjectRestoreInfo3Adapter : IVsProjectRestoreInfo3
+    {
+        public required string BaseIntermediatePath { get; init; }
```

Comment:
Similar throughout, call is MSBuildProjectExtensionsPath. 

Since we're completely reimplementing the API, might as well get the names current (note that I didn't say "correct", cause that BaseIntermediatePath was correct at the point of implementation.).